migrate/live-migration/live-migration-from-postgres.md

```diff
@@ -0,0 +1,252 @@
+[pgcopydb][pgcopydb] to perform a live migration from a source database which
+is using PostgreSQL to Timescale.
+
+Before beginning the migration process, ensure that the following tools are
+installed and available on the system that performs the migration:
+
+- `psql`, `pg_dump`, and `pg_dumpall`: These can be installed by following the
+  instructions in the [How to Install psql on Mac, Ubuntu, Debian, Windows][install-psql]
+  blog post. Although the instructions specifically mention `psql`, following
+  them also installs `pg_dump` and `pg_dumpall`.
+
+- `pgcopydb`: Installation instructions can be found in the
+  [official repository][install-pgcopydb].
+
+<SourceTargetNote />
+
+<!-- TODO:  what about schema changes are they allowed? What other conditions
+are there? -->
+
+In detail, the migration process consists of the following steps:
+
+1. Set up a target database instance in Timescale.
+1. Set up a replication slot and snapshot.
+1. Migrate roles and schema from source to target.
+1. Convert hypertables and enable Timescale features.
+1. Migrate initial data from source to target.
+1. Apply replication changes.
+1. Promote target database as new primary.
+
+<GettingHelp />
+
+<StepOne />
+
+## 2. Set up a replication slot and snapshot
+
+Use the `pgcopydb follow` to create a [replication slot][replication-slot] and
+a [snapshot][snapshot].
+
+```sh
+pgcopydb follow \
+  --source "$SOURCE" \
+  --target "$TARGET" \
+  --fail-fast \
+  --plugin wal2json
+```
+
+<SetupSourceTarget />
+
+The [snapshot][snapshot] ID is exported to `${TMPDIR}/pgcopydb/snapshot`. If
+`$TMPDIR` is not set, it is exported to `/tmp/pgcopydb/snapshot`.
+
+> A snapshot determines which data is visible to the transaction that is using
+the snapshot. Synchronized snapshots are necessary when two or more sessions
+need to see identical content in the database.
+
+The [replication slot][replication-slot] forms the backbone of the replication
+strategy.
+
+> A slot represents a stream of changes that can be replayed to a client in the
+order they were made on the origin server.
+
+The stream of changes emitted by the slot are buffered into disk until they are
+applied on the target. The instance used to orchestrate the migration (the one
+running the commands) should have enough capacity to store the files, and it
+should be actively monitored to prevent any issues that might result due to
+lack of space.
+
+<!-- TODO:  mention the housekeeping script that clears the buffered file s -->
+
+Before the stream of changes can be applied, the schema and data that existed
+prior to the creation of the replication slot must be migrated ([step 3][step-3]). The
+point that marks the beginning of the replication and buffering of changes is
+given by the exported snapshot. The larger the database, the more time it takes
+to perform the initial migration, and the longer the buffered files need to be
+stored.
```

Comment:
I think that we would be well-served by adding a diagram somewhere to explain the overall process. We're working on one for the blog post, which I think we can then put here as well.